lib/reso_transport/query.rb

```diff
@@ -144,6 +152,7 @@ def compile_params
       end
 
       params['$filter'] = compile_filters unless sub_queries.empty?
+      params.merge!(query_parameters) unless query_parameters.empty?
```

Comment:
we are in pure ruby land where we don't have railsy methods like #present?, unless is our best one liner here